aten/src/ATen/native/mobile/cpu/internal/Allocator.cpp

```diff
@@ -0,0 +1,68 @@
+#include <ATen/native/mobile/cpu/internal/Allocator.h>
+
+#ifdef USE_XNNPACK
+
+#include <ATen/native/TensorFactories.h>
+#include <c10/core/CPUAllocator.h>
+
+namespace at {
+namespace native {
+namespace mobile {
+namespace cpu {
+namespace internal {
+
+void Allocator::deleter(void* const memory) {
+  c10::free_cpu(memory);
+}
+
+DataPtr Allocator::allocate(size_t nbytes) const {
+  void* const memory = c10::alloc_cpu(nbytes + kGuard);
+
+  return DataPtr{
+      memory,
+      memory,
+      &deleter,
+      Device(DeviceType::CPU),
+  };
+}
+
+DeleterFnPtr Allocator::raw_deleter() const {
+  return deleter;
+}
+
+Tensor empty(
```

Comment:
Note to @gchanan: we should just have a way to pass in the desired allocator into empty directly...